maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,312 @@ title: Monitor Calico component metrics
+2. In this section you will learn to isolate the resources and create necessary permissions for future steps.
+3. In this section you will configure Prometheus and create its instance in the cluster.
+4. You will be able to visit Prometheus dashboard and create a simple graph.
 
-#### Best practices
+
+### 1. Configure Calico to enable metrics reporting
+#### Felix configuration
+Felix prometheus metrics are **disabled** by default. You have to manually change your Felix configuration (**prometheusMetricsEnabled**) via calicoctl in order to use this feature.
+
+> **Note**: A comprehensive list of configuration values can be [found at this link]({{ site.baseurl }}/reference/felix/configuration).
+   {: .alert .alert-info}
+
+```bash
+calicoctl patch felixConfiguration default  --patch '{"spec":{"prometheusMetricsEnabled": true}}'
+```
+You should see an output like below:
+```
+Successfully patched 1 'FelixConfiguration' resource
+```
+
+##### **Creating a service to expose Felix metrics**
+By using services you will be able to dynamically discover endpoints. Here you will create a service named `felix-metrics-svc` which will receive requests from port 9091 and forward it to port 9090 of pods that are participating in `kube-system` namespace and share `k8s-app: calico-node` label.
+
+> **Note**: Felix by default uses port 9091 TCP to publish its metrics.
+   {: .alert .alert-info}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: felix-metrics-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-node
+  ports:
+  - port: 9091
+    targetPort: 9091
+EOF
+```
+
+#### Typha Configuration
+> **Note** Typha implementation is optional, if you don't have Typha in your cluster you should skip [Typha configuration]({{ site.baseurl }}/maintenance/monitor-component-metrics#typha-configuration) section.
+   {: .alert .alert-danger}
+
+If you are uncertain about `Typha` in your cluster execute the following code:
+
+```bash
+kubectl get pods -A | grep typha
+```
+If your result is similar to what is shown below you are using Typha in your cluster.
+
+> **Note** The name suffix of pods shown below was dynamically generated. Your typha instance might have a different suffix.
+   {: .alert .alert-warning}
+
+```
+kube-system     calico-typha-56fccfcdc4-z27xj                         1/1     Running   0          28h
+kube-system     calico-typha-horizontal-autoscaler-74f77cd87c-6hx27   1/1     Running   0          28h
+```
+
+You can enable Typha metrics to be consumed by Prometheus via [two ways](http://localhost:4000/reference/typha/configuration).
+##### **Creating a service to expose Typha metrics**
+
+By using services you will be able to dynamically discover endpoints. Here you will create a service named `typha-metrics-svc` which will receive requests from port 9091 and forward it to port 9090 of pods that are participating in `kube-system` namespace and share `k8s-app: calico-typha` label.
+
```

Comment:
Something like this?

It is worth mentioning that this service provides endpoints and the port that Typha is using and it will not be used to connect to the instance and collect metrics.